### PR TITLE
refactor Queue.hasFlush: Boolean to Queue.flush: Option[Bool].

### PR DESCRIPTION
**Pull request by sequencer**
Using factory `Queue(..., hasFlush = true)` won't take effects, since in the `Queue.apply` API, `Queue` Module is not exposed, thus even user defines `hasFlush = true`, there is no place for them to give the `flush` signal.
This commit fix this, refactor `Queue.hasFlush: Boolean` to `Queue.flush: Option[Bool]`, makes user be able to pass the flush signal into `Queue` Module.

### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement
- bug fix

#### API Impact

change API Queue.apply API

#### Backend Code Generation Impact
None

#### Desired Merge Strategy
- Squash: The PR will be squashed and merged (choose this if you have no preference. 

#### Release Notes
None

### Reviewer Checklist (only modified by reviewer)
- [x] Did you add the appropriate labels?
- [x] Did you mark the proper milestone (Bug fix: `3.3.x`, [small] API extension: `3.4.x`, API modification or big change: `3.5.0`)?
- [x] Did you review?
- [x] Did you check whether all relevant Contributor checkboxes have been checked?
- [x] Did you mark as `Please Merge`?

### CODE REVIEW

File: src/main/scala/chisel3/util/Decoupled.scala [hunk 1]
```diff
@@ -306,15 +306,16 @@ object Queue
...
     if (entries == 0) {
       val deq = Wire(new DecoupledIO(chiselTypeOf(enq.bits)))
       deq.valid := enq.valid
       deq.bits := enq.bits
       enq.ready := deq.ready
       deq
     } else {
-      val q = Module(new Queue(chiselTypeOf(enq.bits), entries, pipe, flow, useSyncReadMem, hasFlush))
+      val q = Module(new Queue(chiselTypeOf(enq.bits), entries, pipe, flow, useSyncReadMem, flush.isDefined))
+      q.io.flush.foreach(_ := flush.getOrElse(false.B))
```
Comment (mwachs5) [inline src/main/scala/chisel3/util/Decoupled.scala:318]: Do we need the getOrElse? This should not fail if we just use get. Or you could zip them

File: src/main/scala/chisel3/util/Decoupled.scala [hunk 2]
```diff
@@ -306,15 +306,16 @@ object Queue
       pipe: Boolean = false,
       flow: Boolean = false,
       useSyncReadMem: Boolean = false,
-      hasFlush: Boolean = false): DecoupledIO[T] = {
+      flush: Option[Bool] = None): DecoupledIO[T] = {
```
Comment (mwachs5) [inline src/main/scala/chisel3/util/Decoupled.scala:309]: Can you update the scala doc for this function

File: src/main/scala/chisel3/util/Decoupled.scala [hunk 3]
```diff
@@ -333,8 +334,9 @@ object Queue
       entries: Int = 2,
       pipe: Boolean = false,
       flow: Boolean = false,
-      useSyncReadMem: Boolean = false): IrrevocableIO[T] = {
-    val deq = apply(enq, entries, pipe, flow, useSyncReadMem)
+      useSyncReadMem: Boolean = false,
```
Comment (mwachs5) [inline src/main/scala/chisel3/util/Decoupled.scala:337]: Can you update the scaladoc

**Comment by mwachs5** [comment]
What happened to the PR template? The changes look good but can you put the template back?

**Comment by sequencer** [comment]
> What happened to the PR template? The changes look good but can you put the template back?

Oh my bad, I remember I use `gh` to directly pr this, and this looks bad, I will refactor it back.

**Comment by sequencer** [comment]
Done @mwachs5

**Review by mwachs5 — APPROVED** [review]

**Comment by mwachs5** [comment]
flush does not exist on 3.4 so marking this as 3.5

**Comment by Burnleydev1** [comment]
> flush does not exist on 3.4 so marking this as 3.5

Hi @mwachs5, I learned that chisel3.5 will be released tomorrow or Friday.
Please I wish to ask if chisel will be deprecated by then.

**Comment by mwachs5** [comment]
> Please I wish to ask if @chiselName will be deprecated by then.


No, I don't think so, because we haven't worked out the issues around it

**Comment by Burnleydev1** [comment]
> No, I don't think so, because we haven't worked out the issues around it

Okay @mwachs5,  Thank you.

**Pull request merged**